apps/prairielearn/src/pages/administratorCourseRequests/administratorCourseRequests.js

```diff
@@ -18,8 +25,35 @@ router.get(
   '/',
   asyncHandler(async (req, res) => {
     res.locals.coursesRoot = config.coursesRoot;
-    const result = await sqldb.queryOneRowAsync(sql.get_requests, []);
-    _.assign(res.locals, result.rows[0]);
+    const result = await sqldb.queryRow(
+      sql.get_requests,
+      [],
+      z.object({
+        institutions: z.array(InstitutionSchema),
+        course_requests: z.array(
+          CourseRequestSchema.extend({
+            user_name: z.string(),
+            created_at: DateFromISOString.optional(),
```

Comment:
Why is this marked optional? It's a `NOT NULL` column. It's also not clear to me why this is repeated here given that it's already present in `CourseRequestSchema`.

Reply:
The query written was not returning a value for `created_at` and I was using `.optional()` as a way to modify the `CourseRequestSchema` for an undefined value. I have now written `CourseRequestRowSchema` that matches the returned values from the query.

Reply:
Note that you can use [`.pick(...)`](https://zod.dev/?id=pickomit) if you need to construct a schema from another schema but include only a subset of its properties.